### PR TITLE
Add DecomposedFlow rescaler proportional to max AC current load

### PR DESCRIPTION
**Pull request by caioluke**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**
- Refactor DecomposedFlowRescaler API
- Add new DecomposedFlowRescaler based on the maximum current overload
- Retrieve acFlows for both terminals in FlowDecompositionObserver
- Retrieve acCurrents for both terminals in FlowDecompositionObserver

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by caioluke** [comment]
Welcome back, Hugo :)

Yes, indeed this PR adds quite some new stuff in the rescaler.

What I need is to rescale DC flows in such a way that I get the same level of *current* overload as in the AC case.

Therefore, I compare AC current overloads to find the terminal (1 or 2) with the highest current overload, and get the associated active power to rescale DC flows.
In case there are missing limits, I just compare AC currents.

Another subtlety is that I need to calculate these rescaling AC active powers with nominal voltage - as I have done in the code.

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java [hunk 1]
```diff
@@ -26,6 +28,12 @@ public class DecomposedFlow {
     private final double xNodeFlow;
     private final double pstFlow;
     private final double internalFlow;
+    private final double acTerminal1Current;
+    private final double acTerminal2Current;
+    private final double nominalVoltageTerminal1;
+    private final double nominalVoltageTerminal2;
+    private final CurrentLimits currentLimitsTerminal1;
+    private final CurrentLimits currentLimitsTerminal2;
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java:36]: Could you avoid adding these new fields to a `DecomposedFlow` ?

I do not understand why putting them in the `DecomposedFlow` object if you use them once and then never again as you can find those constant information in the branch.

I understand that the AC current might be relevant. 

From a green IT perspective, these nominal voltage and current limits fields are not mandatory for each rescaler.

A flow decomposition on a real networks (Core for example) could lead to a huge number of `DecomposedFlow` in memory. 

Instead of storing these information here, a better strategy might be to modify the API from`DecomposedFlowRescale.rescale(DecomposedFlow decomposedFlow)` to `DecomposedFlowRescale.rescale(DecomposedFlow decomposedFlow, Branch branch)`.

Thus, you will be able to find your required information (nominal voltage, current limit, ...) for each rescaler

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java [hunk 2]
```diff
@@ -49,6 +57,12 @@ protected DecomposedFlow(DecomposedFlowBuilder builder) {
         this.pstFlow = builder.pstFlow;
         this.internalFlow = builder.internalFlow;
         this.loopFlowsMap.putAll(Objects.requireNonNull(builder.loopFlowsMap));
+        this.acTerminal1Current = builder.acCurrentTerminal1;
+        this.acTerminal2Current = builder.acCurrentTerminal2;
+        this.nominalVoltageTerminal1 = builder.nominalVoltageTerminal1;
+        this.nominalVoltageTerminal2 = builder.nominalVoltageTerminal2;
+        this.currentLimitsTerminal1 = builder.currentLimitsTerminal1;
+        this.currentLimitsTerminal2 = builder.currentLimitsTerminal2;
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java:65]: Remove this please

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java [hunk 3]
```diff
@@ -123,6 +137,30 @@ public double getMaxAbsAcFlow() {
...
+    }
+
+    public CurrentLimits getCurrentLimitsTerminal1() {
+        return currentLimitsTerminal1;
+    }
+
+    public CurrentLimits getCurrentLimitsTerminal2() {
+        return currentLimitsTerminal2;
+    }
+
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java:163]: Remove this please

**Review by OpenSuze — CHANGES_REQUESTED** [review]
Thank you for the PR !

Sorry for the delay, I had other projects to attend to...

Could you rename your PR with a less generic name ?

I know that I ask for a lot of modifications. 

I do not what to do another PR to fix yours. 

Feel free to ask more details about this feedback

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java (hunk 1 above)
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java]: Thank you for your suggestions.
I refactored the DecomposedFlowRescaler API, please tell me what you think of it! @OpenSuze

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowComputerUtils.java [hunk 4]
```diff
@@ -34,11 +35,56 @@ public static Map<String, Double> calculateAcTerminalReferenceFlows(Collection<B
         return getTerminalReferenceFlow(xnecList, side);
     }
 
+    public static Map<String, Pair<Double, Double>> getBothTerminalsReferenceFlowsAc(Collection<Branch> xnecList, LoadFlowRunningService.Result loadFlowServiceAcResult) {
```
[the remark this answers is not in the document]
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowComputerUtils.java]: Fair enough, changed it.

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java [hunk 5]
```diff
@@ -48,6 +48,29 @@ public static void assertCoherenceTotalFlow(FlowDecompositionParameters.RescaleM
...
+                    if (currentLimitsTerminal1 == null || currentLimitsTerminal2 == null) {
+                        pActivePowerOnly = acCurrentTerminal1 >= acCurrentTerminal2 ? pTerminal1ActivePowerOnly : pTerminal2ActivePowerOnly;
+                    } else {
+                        double permanentLimitTerminal1 = currentLimitsTerminal1.getPermanentLimit();
+                        double permanentLimitTerminal2 = currentLimitsTerminal2.getPermanentLimit();
+                        double currentOverloadTerminal1 = acCurrentTerminal1 / permanentLimitTerminal1;
+                        double currentOverloadTerminal2 = acCurrentTerminal2 / permanentLimitTerminal2;
+                        pActivePowerOnly = currentOverloadTerminal1 >= currentOverloadTerminal2 ? pTerminal1ActivePowerOnly : pTerminal2ActivePowerOnly;
+                    }
+                    assertEquals(pActivePowerOnly, decomposedFlow.getTotalFlow(), EPSILON);
```
[the remark this answers is not in the document]
Comment (caioluke) [reply flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java]: I added two tests with a fixed result in RescalingTests and left this one as it is.
Would that be ok for you?

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerAcerMethodology.java [hunk 6]
```diff
@@ -34,24 +33,19 @@ private static double rescaleValue(double initialValue, double delta, double sum
...
     @Override
-    public DecomposedFlow rescale(DecomposedFlow decomposedFlow) {
-        double acTerminal1ReferenceFlow = decomposedFlow.getAcTerminal1ReferenceFlow();
-        double acTerminal2ReferenceFlow = decomposedFlow.getAcTerminal2ReferenceFlow();
-        if (Double.isNaN(acTerminal1ReferenceFlow) || Double.isNaN(acTerminal2ReferenceFlow)) {
-            return decomposedFlow;
-        }
+    protected boolean shouldRescaleFlows(DecomposedFlow decomposedFlow) {
+        // - if AC flows are not NaN
+        return !Double.isNaN(decomposedFlow.getAcTerminal1ReferenceFlow()) && !Double.isNaN(decomposedFlow.getAcTerminal2ReferenceFlow());
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerAcerMethodology.java:38]: `Double.isFinite` would be more readable so you can remove the comment

Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerAcerMethodology.java:38]: Extract a static method `hasFiniteAcFlowsOnEachTerminal` to share it with other `DecomposedFlowRescalers`.

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerMaxCurrentOverload.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    public DecomposedFlowRescalerMaxCurrentOverload() {
+        this(DecomposedFlowRescalerProportional.DEFAULT_MIN_FLOW_TOLERANCE);
+    }
+
+    @Override
+    protected boolean shouldRescaleFlows(DecomposedFlow decomposedFlow) {
+        // - if AC flows are not NaN
+        // - if dcReferenceFlow is big enough
+        return !Double.isNaN(decomposedFlow.getAcTerminal1ReferenceFlow()) && !Double.isNaN(decomposedFlow.getAcTerminal2ReferenceFlow()) && Math.abs(decomposedFlow.getDcReferenceFlow()) >= minFlowTolerance;
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerMaxCurrentOverload.java:37]: - Use `hasFiniteAcFlowsOnEachTerminal` 
- Create another static method `hasAbsDcFlowGreaterThanTolerance`

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerNoOp.java [hunk 8]
```diff
@@ -8,20 +8,26 @@
...
     @Override
-    public DecomposedFlow rescale(DecomposedFlow decomposedFlow) {
-        return decomposedFlow;
+    protected boolean shouldRescaleFlows(DecomposedFlow decomposedFlow) {
+        return false;
+    }
+
+    @Override
+    protected RescaledFlows computeRescaledFlows(DecomposedFlow decomposedFlow, Network network) {
+        return null;
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerNoOp.java:31]: Maybe it should raise an exception has by design we cannot rescale with such rescaler

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java [hunk 9]
```diff
@@ -30,53 +29,25 @@ public DecomposedFlowRescalerProportional() {
...
-        Country country2 = decomposedFlow.getCountry2();
-        double allocatedFlow = decomposedFlow.getAllocatedFlow();
-        double xNodeFlow = decomposedFlow.getXNodeFlow();
-        double pstFlow = decomposedFlow.getPstFlow();
-        double internalFlow = decomposedFlow.getInternalFlow();
-        Map<String, Double> loopFlows = decomposedFlow.getLoopFlows();
+    protected boolean shouldRescaleFlows(DecomposedFlow decomposedFlow) {
+        // - if AC flows are not NaN
+        // - if dcReferenceFlow is big enough
+        return !Double.isNaN(decomposedFlow.getAcTerminal1ReferenceFlow()) && !Double.isNaN(decomposedFlow.getAcTerminal2ReferenceFlow()) && Math.abs(decomposedFlow.getDcReferenceFlow()) >= minFlowTolerance;
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java:35]: Reuse static functions here please

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java [hunk 10]
```diff
@@ -8,16 +8,15 @@
...
+import com.powsybl.iidm.network.Network;
 
 import java.util.Map;
 import java.util.stream.Collectors;
 
 /**
  * @author Caio Luke {@literal <caio.luke at artelys.com>}
  */
-public class DecomposedFlowRescalerProportional implements DecomposedFlowRescaler {
+public class DecomposedFlowRescalerProportional extends AbstractDecomposedRescaler {
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java:19]: Maybe we should move `DEFAULT_MIN_FLOW_TOLERANCE` to `FlowDecompositionParameters` to reverse the dependency direction

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/RescalingTests.java [hunk 11]
```diff
@@ -238,8 +251,79 @@ void testRescalingProportionalDoesNotRescaleWithSmallFlow() {
...
+    }
+
+    @Test
+    void testRescalingMaxCurrentOverloadWithoutCurrentLimits() {
+        Network network = TestUtils.importNetwork("19700101_0000_FO4_UX1.uct");
+        DecomposedFlow decomposedFlow = getDecomposedFlowForMaxCurrentOverload("BB000011 BB000021 1");
+        DecomposedFlow decomposedFlowRescaled = new DecomposedFlowRescalerMaxCurrentOverload().rescale(decomposedFlow, network);
+        final double expectedRescaledInternalFlow = -47.63139720814412;
+        assertTrue(Math.abs(decomposedFlowRescaled.getInternalFlow() - expectedRescaledInternalFlow) < 1E-6);
+    }
```
Comment (OpenSuze) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/RescalingTests.java:328]: Perfect ! Thank you

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java [hunk 12]
```diff
@@ -42,12 +42,34 @@ public static GlskDocument importGlskDocument(String glskFileName) {
...
+
+                    double pActivePowerOnly;
+                    if (currentLimitsTerminal1 == null || currentLimitsTerminal2 == null) {
+                        pActivePowerOnly = acCurrentTerminal1 >= acCurrentTerminal2 ? pTerminal1ActivePowerOnly : pTerminal2ActivePowerOnly;
+                    } else {
+                        double currentOverloadTerminal1 = acCurrentTerminal1 / currentLimitsTerminal1.getPermanentLimit();
+                        double currentOverloadTerminal2 = acCurrentTerminal2 / currentLimitsTerminal2.getPermanentLimit();
+                        pActivePowerOnly = currentOverloadTerminal1 >= currentOverloadTerminal2 ? pTerminal1ActivePowerOnly : pTerminal2ActivePowerOnly;
+                    }
+                    assertEquals(pActivePowerOnly, decomposedFlow.getTotalFlow(), EPSILON);
```
Comment (OpenSuze) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java:71]: I'm still not a huge fan of duplicating code here to check coherence. 
I would rather throw an exception that duplicating code...

**Review by OpenSuze — CHANGES_REQUESTED** [review]
Thank you for your changes ! I appreciate your new `DecomposedFlowRescaler` API :)

**Comment by OpenSuze** [comment]
Could you add a bit of documentation please ?

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionObserver.java [hunk 13]
```diff
@@ -91,14 +91,42 @@ public interface FlowDecompositionObserver {
...
+     */
+    void computedDcFlowsTerminal1(Map<String, Double> flows);
 
     /**
      * Called when the DC loadflow is computed (for base case or contingency)
      *
-     * @param flows the flows for all branches
+     * @param flows the terminal 2 flow for all branches
+     */
+    void computedDcFlowsTerminal2(Map<String, Double> flows);
```
[the remark this answers is not in the document]
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionObserver.java:117]: Indeed, you are right. I reverted back to dcFlows.

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/AbstractDecomposedRescaler.java
[the remark this answers is not in the document]
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/AbstractDecomposedRescaler.java:1]: It was your idea, I'm only following orders :)

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/AbstractDecomposedRescaler.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                .withAcCurrentTerminal2(decomposedFlow.getAcTerminal2Current())
+                .withAllocatedFlow(rescaledFlows.rescaledAllocatedFlow())
+                .withXNodeFlow(rescaledFlows.rescaledXNodeFlow())
+                .withPstFlow(rescaledFlows.rescaledPstFlow())
+                .withInternalFlow(rescaledFlows.rescaleInternalFlow())
+                .withLoopFlowsMap(rescaledFlows.rescaledLoopFlows())
+                .build();
+    }
+
+    protected record RescaledFlows(double rescaledAllocatedFlow, double rescaledXNodeFlow, double rescaledPstFlow, double rescaleInternalFlow, Map<String, Double> rescaledLoopFlows) { }
```
[the remark this answers is not in the document]
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/AbstractDecomposedRescaler.java]: Thanks!

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java (hunk 10 above)
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java]: Which dependency direction?

**Comment by caioluke** [comment]
> Could you add a bit of documentation please ?

I added some documentation.
I was wondering what to write about the parameter `proportional-rescaler-min-flow-tolerance`, since now it's also used by the new Rescaler...
Do you have a suggestion? Maybe change it's name?

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionObserverTest.java [hunk 15]
```diff
@@ -295,12 +317,12 @@ void testNStateN1AndN2PostContingencyState() {
...
         // Checking AC flows
         for (var contingencyId : List.of(BASE_CASE, contingencyId1, contingencyId2)) {
-            assertEquals(allBranches, report.acFlows.forContingency(contingencyId).keySet());
+            assertEquals(allBranches, report.acFlowsTerminal1.forContingency(contingencyId).keySet());
         }
 
         // Checking DC flows
         for (var contingencyId : List.of(BASE_CASE, contingencyId1, contingencyId2)) {
-            assertEquals(allBranches, report.dcFlows.forContingency(contingencyId).keySet());
+            assertEquals(allBranches, report.acFlowsTerminal1.forContingency(contingencyId).keySet());
```
Comment (OpenSuze) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionObserverTest.java:325]: `dcFlows` instead of `acFlowsTerminal1` no ?

**Review by OpenSuze — CHANGES_REQUESTED** [review]

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java (hunk 10 above)
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java]: We should move DEFAULT_MIN_FLOW_TOLERANCE to FlowDecompositionParameters.

Thus the rescalers default tolerance will depend on the FlowDecompositionParameters

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (OpenSuze) [reply flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java]: Great, thank you !

**Comment by OpenSuze** [comment]
> > Could you add a bit of documentation please ?
> 
> @OpenSuze I added some documentation. I was wondering what to write about the parameter `proportional-rescaler-min-flow-tolerance`, since now it's also used by the new Rescaler... Do you have a suggestion? Maybe change it's name?

Thank you for the documentation !

I think that I am ok with the parameter name. Both rescalers have some kind of proportionality.

File: docs/flow_decomposition/algorithm-description.md [hunk 16]
```diff
@@ -127,6 +127,20 @@ The difference between reference AC flow and the sum of the parts of the decompo
...
+In case there are missing limits, we just compare AC currents instead of overloads.
+Hence, the coefficient is defined as
+1) With current limits:
+- if $$\frac{AC I1}{I_{max}1} >= \frac{AC I2}{I_{max}2}$$, then $$\alpha_{\text{rescale}} = \frac{\sqrt{3} \cdot AC I1 \cdot \frac{V1_{nominal}}{1000}}{|DC p1|}$$
+- else, $$\alpha_{\text{rescale}} = \frac{\sqrt{3} \cdot AC I2 \cdot \frac{V2_{nominal}}{1000}}{|DC p1|}$$
+2) Without current limits:
+- if $$AC I1 >= AC I2$$, then $$\alpha_{\text{rescale}} = \frac{\sqrt{3} \cdot AC I1 \cdot \frac{V1_{nominal}}{1000}}{|DC p1|}$$
+- else, $$\alpha_{\text{rescale}} = \frac{\sqrt{3} \cdot AC I2 \cdot \frac{V2_{nominal}}{1000}}{|DC p1|}$$
+
+Since we divide by the DC flow to calculate the coefficient, lines with a too small DC flow are not rescaled.
```
[the remark this answers is not in the document]
Comment (caioluke) [reply docs/flow_decomposition/algorithm-description.md]: Indeed, well spotted

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionObserverTest.java (hunk 15 above)
Comment (caioluke) [reply flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionObserverTest.java]: Ops, thanks

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java (hunk 10 above)
Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescalerProportional.java]: Ah, I get it now! I must've been tired on friday, I didn't realize you were talking about the constant.
Yes, it makes total sense to move it there.
I just did it.

**Review by OpenSuze — APPROVED** [review]
Thank you for your PR !

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescaler.java [hunk 17]
```diff
@@ -8,12 +8,13 @@
...
+import com.powsybl.iidm.network.Network;
 
 /**
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  * @author Hugo Schindler {@literal <hugo.schindler at rte-france.com>}
  * @author Caio Luke {@literal <caio.luke at artelys.com>}
  */
 public interface DecomposedFlowRescaler {
-    DecomposedFlow rescale(DecomposedFlow decomposedFlow);
+    DecomposedFlow rescale(DecomposedFlow decomposedFlow, Network network);
```
Comment (OpenSuze) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescaler.java:19]: We have a last suggestion:
```suggestion
    default DecomposedFlow rescale(DecomposedFlow flow) {
        return rescale(flow, null);
    }

    DecomposedFlow rescale(DecomposedFlow decomposedFlow, Network network);
```
This will avoid to add a breaking change.
You can also revert the change in the implementation that does not need a `network`.
This will also avoid using `null` in some tests.

Comment (caioluke) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescaler.java:24]: Well spotted :)

> You can also revert the change in the implementation that does not need a network.
I'm not sure I understand, do I still need to change something else?

**Review by OpenSuze — APPROVED** [review]

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescaler.java (hunk 17 above)
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/rescaler/DecomposedFlowRescaler.java:24]: You did well ! I mean to revert changes in tests (not the implementation)
